server/src/main/java/io/crate/execution/ddl/tables/AddColumnTask.java

```diff
@@ -101,6 +101,26 @@ protected ClusterState execute(ClusterState currentState, AddColumnRequest reque
 
         request = addMissingParentColumns(request, currentTable);
         HashMap<ColumnIdent, List<Reference>> tree = buildTree(request.references());
+
+        // Request can contain parent columns of the new columns in order to be able to build and traverse the tree.
+        // For incrementing column OID we only count new columns which is basically equivalent of counting leaves.
+        // We set oid to References so that buildMapping method below would actually persist it in the mapping.
+        long currentOID = currentState.metadata().columnOID();
+        for (Reference reference: request.references()) {
+            // Ref is a leaf if it doesn't have children.
+            if (tree.get(reference.column()) == null) {
+                if (reference instanceof GeneratedReference generatedReference) {
+                    SimpleReference simpleReference = (SimpleReference) generatedReference.reference();
```

Comment:
todo: test/clarify whether we can have multiple levels of generated refs. then would need smth like 
```
while(cureRef instance of GenRef genRef) {
curRef = genRef.reference();
}
```